jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -954,6 +954,21 @@ public Collection<Violation> getViolations()
             return _violations == null ? Collections.emptySet() : Collections.unmodifiableCollection(_violations);
         }
 
+        /**
+         * Clear, from the violations list, any path related violations.
+         */
+        private void clearPathViolations()
+        {
+            if (_violations == null)
+                return;
+            _violations.removeIf((violation) ->
+                    (violation == Violation.AMBIGUOUS_PATH_PARAMETER) ||
+                    (violation == Violation.AMBIGUOUS_PATH_SEGMENT) ||
+                    (violation == Violation.AMBIGUOUS_PATH_SEPARATOR) ||
+                    (violation == Violation.AMBIGUOUS_PATH_ENCODING) ||
+                    (violation == Violation.AMBIGUOUS_EMPTY_SEGMENT));
```

Comment:
Can you move this to a `UrlCompliance.isPathViolation(Violation)` method?    
Having it in the same class will make it a little less fragile to the addition of new violations.

Reply:
Done.